cpp/src/arrow/ipc/test_common.cc

```diff
@@ -1000,6 +1003,102 @@ Status MakeDictExtension(std::shared_ptr<RecordBatch>* out) {
   return Status::OK();
 }
 
+namespace {
+
+template <typename CValueType, typename SeedType, typename DistributionType>
+void FillRandomData(CValueType* data, size_t n, CValueType min, CValueType max,
+                    SeedType seed) {
+  std::default_random_engine rng(seed);
+  DistributionType dist(min, max);
+  std::generate(data, data + n,
+                [&dist, &rng] { return static_cast<CValueType>(dist(rng)); });
+}
+
+template <typename CValueType, typename SeedType>
+enable_if_t<std::is_integral<CValueType>::value && std::is_signed<CValueType>::value,
+            void>
+FillRandomData(CValueType* data, size_t n, SeedType seed) {
+  FillRandomData<CValueType, SeedType, std::uniform_int_distribution<CValueType>>(
+      data, n, 0, 1000, seed);
+}
+
+template <typename CValueType, typename SeedType>
+enable_if_t<std::is_integral<CValueType>::value && std::is_unsigned<CValueType>::value,
+            void>
+FillRandomData(CValueType* data, size_t n, SeedType seed) {
+  FillRandomData<CValueType, SeedType, std::uniform_int_distribution<CValueType>>(
+      data, n, -1000, 1000, seed);
```

Comment:
Looks like you're using -1000 as min for an unsigned type?

Reply:
This function is specialized for unsigned integer types by the `enable_if_t` at line 1026.
The function for singed integer types is defined before this function.

Reply:
This is what I said: you're using a negative value in a function specialized for unsigned ints... WIll change.